### PR TITLE
Update current DatePicker value in UI

### LINKED ISSUE #9834: [BUG] Cannot update DatePicker display value from Python
#### ALL software version info (bokeh, python, notebook, OS, browser, any other relevant packages)
Bokeh 2.0.0

#### Description of expected behavior and the observed behavior
The initial report at https://discourse.bokeh.org/t/datepicker-ui-not-updating-in-bokeh-2-0/5021/2

#### Complete, minimal, self-contained example code that reproduces the issue

```python
from bokeh.plotting import curdoc
from bokeh.layouts import column
from bokeh.models.widgets import Button, DatePicker
from datetime import date
dp = DatePicker(title='Date')
dp.value = date(2020, 1, 3)
print(dp, dp.value)
bt = Button(label='button')
def update_dp():
       print(dp, dp.value)
       dp.value = date(2020, 2, 2)
bt.on_click(update_dp)
curdoc().add_root(column(dp,bt))
```

### PR DESCRIPTION
**Pull request by p-himik**
- [x] issues: fixes #9834
- [ ] tests added / passed
- [ ] release document entry (if new feature or API change)

### CODE REVIEW

File: bokehjs/src/lib/models/widgets/date_picker.ts [hunk 1]
```diff
@@ -39,7 +39,11 @@ export class DatePickerView extends InputWidgetView {
     super.connect_signals()
 
     const {value, min_date, max_date, disabled_dates, enabled_dates, position, inline} = this.model.properties
-    this.connect(value.change, () => this._set("defaultDate", value.value()))
+    this.connect(value.change, () => {
+      if (this._picker != null) {
+        this._picker.setDate(value.value())
+      }
```
Comment (mattpap) [inline bokehjs/src/lib/models/widgets/date_picker.ts]: This could be simplified with modern ES syntax:
```ts
this._picker?.setDate(value.value())
```

Comment (p-himik) [reply bokehjs/src/lib/models/widgets/date_picker.ts]: Would it make sense to replace the `_set` private method as well?

Comment (mattpap) [reply bokehjs/src/lib/models/widgets/date_picker.ts]: I would drop `_set()` completely, especially given it is type unsafe (e.g. uses `string` key)

**Comment by bryevdv** [comment]
@p-himik  would you want to try to add an integration test? The relevant file is here:

https://github.com/bokeh/bokeh/blob/master/tests/integration/widgets/test_datepicker.py

**Comment by p-himik** [comment]
@bryevdv Tried to, but for some reason the tests don't work with `geckodriver` and `chromedriver` keeps crashing on my system.

This is the code that I've come up so far:
```python
    @flaky(max_runs=10)
    def test_server_value_change(self, bokeh_server_page) -> None:
        new_date = '2019-9-21'

        def modify_doc(doc):
            dp = DatePicker(value=date(2019, 9, 20), min_date=date(2019, 9, 1), max_date=date(2019, 9, 30))
            b = Button(label='Set date via server', css_classes=['foo'])

            def set_date():
                dp.value = new_date

            b.on_click(set_date)
            doc.add_root(column(dp, b))

        page = bokeh_server_page(modify_doc)

        button_el = page.driver.find_element_by_css_selector('.foo button')
        button_el.click()

        el = page.driver.find_element_by_css_selector('.bk-input')
        assert el.get_attribute('value') == new_date
```

I don't think I can take it any further than that without spending hours upon hours figuring out the issues with the drivers.

**Comment by bryevdv** [comment]
@p-himik That looks pretty reasonable to me, I can test it out but not before I need to get another rc/release ready. For the sake of expediency why don't you open a follow-on issue to add a test and include that WIP code. I will merge this now.

**Pull request merged**